com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ExtenderUtil.java

```diff
@@ -803,49 +803,73 @@ public static void storeResources(File targetDirectory, Map<String, IResource> r
         }
     }
 
-    private static boolean isBinZip(String resourcePath) {
-        return resourcePath.endsWith("plugins/bin/common.zip") || Stream
+    private static boolean isPluginZip(String resourcePath) {
+        return resourcePath.endsWith("plugins/common.zip") || Stream
                 .of(Platform.X86_64MacOS, Platform.X86_64Linux, Platform.X86_64Win32)
                 .flatMap(platform -> Stream.of(platform.getPair(), platform.getOs()))
-                .anyMatch(platformOption -> resourcePath.endsWith("plugins/bin/" + platformOption + ".zip"));
+                .anyMatch(platformOption -> resourcePath.endsWith("plugins/" + platformOption + ".zip"));
     }
 
-    public static void storeResources(File targetDirectory, List<IResource> resources) throws CompileExceptionError {
-        for (IResource resource : resources) {
-            String relativePath = resource.getPath();
-            File outputFile = new File(targetDirectory, relativePath);
-            if (!outputFile.getParentFile().exists()) {
-                outputFile.getParentFile().mkdirs();
-            }
+    private static int getPluginZipSpecificity(String resourcePath) {
+        String baseName = FilenameUtils.getBaseName(resourcePath);
+        if (baseName.equals("common")) {
+            return 0;
+        } else if (baseName.contains("-")) {
+            return 2;
+        } else {
+            return 1;
+        }
+    }
 
-            // We do this because we might be called from within the editor, which ight have the .dll's locked.
-            // and so we don't want to fail copying those if we can avoid it.
-            if (outputFile.exists() && areFilesIdentical(resource, outputFile)) {
-                continue;
+    private static File storeResource(File targetDirectory, IResource resource) throws CompileExceptionError {
+        String relativePath = resource.getPath();
+        File outputFile = new File(targetDirectory, relativePath);
+        if (!outputFile.getParentFile().exists()) {
+            outputFile.getParentFile().mkdirs();
+        }
+
+        // We do this because we might be called from within the editor, which might have the .dll's locked.
+        // and so we don't want to fail copying those if we can avoid it.
+        if (outputFile.exists() && areFilesIdentical(resource, outputFile)) {
+            return outputFile;
+        }
+
+        try {
+            byte[] data = resource.getContent();
+            FileUtils.writeByteArrayToFile(outputFile, data);
+            String outputPath = outputFile.toString();
+            if (outputPath.contains("/plugins/bin/")) {
+                outputFile.setExecutable(true);
             }
+            return outputFile;
+        } catch (Exception e) {
+            throw new CompileExceptionError(resource, 0, e);
+        }
+    }
```

Comment:
I wonder if we could somehow highlight these in the future.
e.g. 
1) show a list of unsigned/external executables that are downloaded from dependencies.
2) we store a cache with executable name/version and if it's a new version, the user has to accept "yes" in order to run it.

Might be overkill, but perhaps work talking about in the future.